services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,310 @@
+	}
+	replicasPerFailureDomain := 1
+	if arbiter {
+		replicasPerFailureDomain = 2
+	}
+
+	cephBlockPool := &cephv1.CephBlockPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      poolName,
+			Namespace: namespace,
+		},
+		Spec: cephv1.NamedBlockPoolSpec{
+			PoolSpec: cephv1.PoolSpec{
+				FailureDomain:  failureDomain,
+				EnableRBDStats: true,
+				DeviceClass:    lsoStorageClassName,
+				Replicated: cephv1.ReplicatedSpec{
+					Size:                     uint(dataProtectionPolicy),
+					RequireSafeReplicaSize:   true,
+					ReplicasPerFailureDomain: uint(replicasPerFailureDomain),
+				},
+				Parameters: map[string]string{
+					"compression_mode": compression,
+				},
+				EnableCrushUpdates: true,
+			},
+		},
+	}
+
+	mutateFunc := func() error {
+		return nil
+	}
+	_, err := ctrl.CreateOrUpdate(r.Context(), client, cephBlockPool, mutateFunc)
+	if err != nil {
+		klog.Errorf("failed to create cephBlockPool: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func createCephBlockPoolStorageClass(w http.ResponseWriter, r *http.Request, client client.Client, storageClassName, poolName, reclaimPolicy, volumeBindingMode string, enableEncryption bool, encryptionKMSID, namespace string) {
+	allowVolumeExpansion := true
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: storageClassName,
+			Annotations: map[string]string{
+				"description": "Provides RWO Filesystem volumes, and RWO and RWX Block volumes",
+				"reclaimspace.csiaddons.openshift.io/schedule": "@weekly",
+			},
+		},
+		Provisioner:       storagecluster.RbdDriverName,
+		ReclaimPolicy:     (*corev1.PersistentVolumeReclaimPolicy)(&reclaimPolicy),
+		VolumeBindingMode: (*storagev1.VolumeBindingMode)(&volumeBindingMode),
+		// AllowVolumeExpansion is set to true to enable expansion of OCS backed Volumes
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Parameters: map[string]string{
+			"clusterID":                 namespace,
+			"pool":                      poolName,
+			"imageFeatures":             "layering,deep-flatten,exclusive-lock,object-map,fast-diff",
+			"csi.storage.k8s.io/fstype": "ext4",
+			"imageFormat":               "2",
+			"encrypted":                 strconv.FormatBool(enableEncryption),
+			"csi.storage.k8s.io/provisioner-secret-name":            "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/provisioner-secret-namespace":       namespace,
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-rbd-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": namespace,
+		},
+	}
+	if enableEncryption {
+		storageClass.Parameters["encryptionKMSID"] = encryptionKMSID
+	}
+
+	mutateFunc := func() error {
+		return nil
+	}
+	_, err := ctrl.CreateOrUpdate(r.Context(), client, storageClass, mutateFunc)
```

Comment:
Since storageClass params cannot be updated after it is created once, would it make sense to use create here? and if the params differ then delete and create it again?